commons/src/main/java/org/infinispan/commons/api/AsyncLifecycle.java

```diff
@@ -0,0 +1,16 @@
+package org.infinispan.commons.api;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Async Lifecycle interface that defines the lifecycle of components in a non blocking way
+ *
+ * @author Katia Aresti (<a href="mailto:karesti@redhat.com">karesti@redhat.com</a>)
+ * @since 9.1
+ */
+public interface AsyncLifecycle {
```

Comment:
No, I didn't implement async ping because in the PR I understood that, as the start can be done in a call just after passing false, we just need something simple right now concerning the start method that is blocking. 

I created the interface because I wanted to add an async method interface for start, as we do in cache. But the lifecycle interface is used a lot and I was just providing something small to cover the request as I understood it on Galder's issue.